overlay/generic/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -141,52 +222,221 @@ is_net()
 	[ $c -lt 0 ] && return 1
 	[ $c -gt 255 ] && return 1
 	[ $d -lt 0 ] && return 1
-	# Make sure the last field isn't the broadcast addr.
-	[ $d -ge 255 ] && return 1
+	return 0
+}
+
+# Tests if input is an email address
+is_email() {
+	regex="^[a-z0-9!#\$%&'*+/=?^_\`{|}~-]+(\.[a-z0-9!#$%&'*+/=?^_\`{|}~-]+)*@([a-z0-9]([a-z0-9-]*[a-z0-9])?\.?)+[a-z0-9]([a-z0-9-]*[a-z0-9])?\$"
+	ADDRESS=$1
+
+	[[ $ADDRESS =~ $regex ]] && return 0
```

Comment:
the `return` statements are redundant here, whatever the return code of the last command ran in a bash function will be the return code of the entire function... 

``` bash
is_email() {
  local regex='...'
  local address=$1
  [[ $address =~ $regex ]]
}
```